openupgrade_scripts/scripts/account/15.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,429 @@
+                    WHERE atpl.id = aml.tax_repartition_line_id) = 'sale'
+                THEN (SELECT refund_tax_id
+                    FROM account_tax_repartition_line
+                    WHERE id = aml.tax_repartition_line_id) IS NULL
+                ELSE false
+            END
+        FROM account_move am
+        WHERE am.id = aml.move_id AND am.move_type = 'entry'
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_move_line aml
+        SET tax_tag_invert =
+            CASE
+                WHEN (SELECT at.type_tax_use
+                    FROM account_move_line_account_tax_rel
+                    JOIN account_tax at ON at.id = account_tax_id
+                    WHERE aml.id = account_move_line_id
+                    LIMIT 1) = 'purchase'
+                THEN aml.credit > 0
+                WHEN (SELECT at.type_tax_use
+                    FROM account_move_line_account_tax_rel
+                    JOIN account_tax at ON at.id = account_tax_id
+                    WHERE aml.id = account_move_line_id
+                    LIMIT 1) = 'sale'
+                THEN aml.debit > 0
+            END
+        FROM account_move am
+        WHERE am.id = aml.move_id AND
+        am.move_type = 'entry' AND
+        aml.tax_tag_invert IS NULL AND
+        (SELECT COUNT(account_tax_id)
+            FROM account_move_line_account_tax_rel
+            WHERE aml.id = account_move_line_id) > 0""",
+    )
+
+
+def _create_account_payment_method_line(env):
+    # Create account_payment_method_line table
+    openupgrade.logged_query(
+        env.cr,
+        """
+        CREATE TABLE account_payment_method_line (
+            id SERIAL,
+            journal_id INTEGER,
+            name varchar,
+            payment_account_id INTEGER,
+            payment_method_id INTEGER NOT NULL,
+            sequence INTEGER,
+            CONSTRAINT account_payment_method_line_pkey PRIMARY KEY (id)
+        )""",
+    )
+    # Create account payment method lines from account payment methods
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO account_payment_method_line
+        (name, payment_method_id, journal_id)
+        SELECT apm.name, apm.id, aj.id
+        FROM account_payment_method apm
+        JOIN account_journal aj ON aj.type IN ('bank', 'cash')
+        WHERE apm.code = 'manual'
+        """,
+    )
+
+
+def _fast_fill_account_payment_payment_method_line_id(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_payment
+        ADD COLUMN IF NOT EXISTS payment_method_line_id INTEGER""",
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_payment ap
+        SET payment_method_line_id = apml.id
+        FROM account_move am
+        JOIN account_payment_method_line apml ON apml.journal_id = am.journal_id
+        WHERE ap.move_id = am.id
```

Comment:
```suggestion
        UPDATE account_payment ap
        SET payment_method_line_id = apml.id
        FROM account_move am
        JOIN account_payment_method_line apml ON apml.journal_id = am.journal_id
        WHERE ap.move_id = am.id AND ap.payment_method_id = apml.payment_method_id
```
Assure that the payment method on payment are mapped with the payment method on the newly created payment method line. In other cases, if the payment method is `check_printing`, the line should be filled by migrating module `account_check_printing`, and if the payment method is `electronic`, should be filled by the `payment_xxx` modules.